pkgs/development/libraries/gsl-lite/default.nix

```diff
@@ -0,0 +1,45 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, ninja
+, installCompatHeader ? false
+, installLegacyHeaders ? false
+}:
+
+stdenv.mkDerivation rec {
+  pname = "gsl-lite";
+  version = "0.40.0";
+
+  src = fetchFromGitHub {
+    owner = "gsl-lite";
+    repo = "gsl-lite";
+    rev = "v${version}";
+    hash = "sha256-80ksT8XFn2LLMr63gKGZD/0+FDLnAtFyMpuuSjtoBlk=";
+  };
+
+  nativeBuildInputs = [ cmake ninja ];
+
+  cmakeFlags = let toggleBool = x: if x then "ON" else "OFF"; in [
+    "-DGSL_LITE_OPT_INSTALL_COMPAT_HEADER:BOOL=${toggleBool installCompatHeader}"
+    "-DGSL_LITE_OPT_INSTALL_LEGACY_HEADERS:BOOL=${toggleBool installLegacyHeaders}"
+  ];
```

Comment:
This is a minor nitpick, but this could be
```nix
lib.attrsets.mapAttrsToList
  (name: value: ''DGSL_LITE_OPT_${name}:BOOL=${if value then "ON" else "OFF"}'')
  { INSTALL_COMPAT_HEADER = installCompatHeader;
    NSTALL_LEGACY_HEADERS = installLegacyHeaders;
  }
```

Reply:
I've moved this package to #196121.